include/singa/core/device.h

```diff
@@ -51,6 +51,17 @@ using std::vector;
 
 namespace singa {
 
+/// Method used by time profiling
+/// ProfilingMode == 0 (default) -> Chrono
+/// ProfilingMode == 1 -> Event (e.g. cudaEvent)
+enum ProfilingMode { useChrono, useEvent };
```

Comment:
I think chorno is only applicable for measuring cpu time and cudaevent is for measuring gpu time.
e.g., we cannot use cudaEvent if we are running on cpu.

Reply:
yes, cpu can use chrono only, gpu can use either 1. chrono or 2. cudaevent

Time Profiling methods in NVIDIA documentation:
1. Chrono: https://docs.nvidia.com/deeplearning/sdk/tensorrt-best-practices/index.html#cpu-timing
2. CudaEvent: https://docs.nvidia.com/deeplearning/sdk/tensorrt-best-practices/index.html#cuda-events
I have implemented both for GPU

Reply:
I think nvidia suggests to use chrono for cpu and use cudaevent for gpu operations because chorno may not measure the time correctly due to asynchronous execution; 
then we do not need users to select the mode.

Reply:
ok, I delete the switch between the modes

Reply:
I will modify the code to remove the switch

Reply:
I have removed the profiling mode, now cpu use chrono, gpu use cudaevent